tensor_genn/algorithms/relu_ann.py

```diff
@@ -6,21 +6,31 @@
 
 class ReLUANN():
     def __init__(self,neuron_resting_voltage=-60.0,neuron_threshold_voltage=-55.0,
-        membrane_capacitance=1.0, model_timestep=1.0,):
+        membrane_capacitance=1.0, model_timestep=1.0, single_example_time=350.):
         self.Vres = neuron_resting_voltage
         self.Vthr = neuron_threshold_voltage
         self.Cm = membrane_capacitance
         self.timestep = model_timestep
+        self.single_example_time = single_example_time
 
     def convert(self, tf_model):
+        # Check model compatibility
+        if not isinstance(tf_model,tf.keras.models.Sequential):
+            raise NotImplementedError('Implementation for type {} models not found'.format(type(tf_model)))
+        
+        for layer in tf_model.layers:
+            if not isinstance(layer,(tf.keras.layers.Dense,tf.keras.layers.Flatten)):
+                raise NotImplementedError('Only Dense layers are supported')
+        if len(tf_model.get_weights()) >= len(tf_model.layers):
```

Comment:
How does this check work?

Reply:
The Dense layer check just loops through each layer and checks the type.
Again (like activation fn), there is no direct way to access the use_bias parameter. So this checks if the number of weight matrices is greater than the number of synapses. If so, it means that atleast one synapse has more than 1 weight matrix i.e a bias exists too. 
Does this make sense?

Reply:
I just had a quick look and you can access the ``activation`` and ``use_bias`` properties on ``Dense`` layers but not ``Flatten`` layers as they don't have them - I think it would be nicer to check the type of each layer in the loop as you encounter them and use these properties

Reply:
I've implemented the checks for all layers except the last one since it may be used for classification and hence may use softmax/sigmoid